src/containers/Layout/index.tsx

```diff
@@ -1,25 +1,48 @@
+import React from 'react';
+
 import { PAGE_CONTAINER_ID } from 'constants/layout';
 
 import { Footer } from './Footer';
 import { Header } from './Header';
 import { Menu } from './Menu';
+import ScrollToTop from './ScrollToTop';
+import { store } from './store';
 
 export interface LayoutProps {
   children?: React.ReactNode;
 }
 
-export const Layout: React.FC<LayoutProps> = ({ children }) => (
-  <div className="flex h-screen flex-col md:flex-row">
-    <Menu />
+export const Layout: React.FC<LayoutProps> = ({ children }) => {
+  const setScrollToTopVisible = store.use.setScrollToTopVisible();
+  const viewportHeight = window.innerHeight;
+  const handleScroll = (event: React.UIEvent<HTMLDivElement>) => {
+    const scrollElem = event.currentTarget;
+    if (scrollElem?.scrollTop && scrollElem.scrollTop > viewportHeight) {
+      setScrollToTopVisible(true);
+    } else {
+      setScrollToTopVisible(false);
+    }
+  };
```

Comment:
That's going to trigger a lot of re-renders, because on every scroll event the store state gets updated. You can limit the amount of updates/re-renders by only updating the store's state when needed:
```suggestion
  const isScrollToTopVisible = store.use. isScrollToTopVisible();

  const handleScroll = (event: React.UIEvent<HTMLDivElement>) => {
    const scrollElem = event.currentTarget;
    const shouldShowScrollToTopVisible = scrollElem?.scrollTop && scrollElem.scrollTop > viewportHeight;
    
    if (shouldShowScrollToTopVisible && !isScrollToTopVisible) {
      setScrollToTopVisible(true);
    } else if (!shouldShowScrollToTopVisible && isScrollToTopVisible) {
      setScrollToTopVisible(false);
    }
  };
```